dev/com.ibm.ws.security.mp.jwt.1.1.config/src/com/ibm/ws/security/mp/jwt/v11/config/impl/MpConfigProxyServiceImpl.java

```diff
@@ -77,6 +77,36 @@ public <T> T getConfigValue(ClassLoader cl, String propertyName, Class<T> proper
         return null;
     }
 
+    /** return */
+    @Sensitive
+    @Override
+    public MpConfigProperties getConfigProperties(ClassLoader cl) {
```

Comment:
It would be awesome to do that, but the interface where the default method is does not have the ability to reference MP Config APIs.  I think the only possible option is to have another project that has an abstract version of the class that the others can all pull into their bundles if you want to avoid the code duplication.

Reply:
I guess the other option is to have an interface that has an implementation that wraps the Config object and there is a method that returns an impl of that interface in each of the 3 impls.

Reply:
It would look something like this:
```
    interface MpConfigProxy {
        <T> Optional<T> getOptionalValue(String propertyName, Class<T> propertyType);
    }

    public MpConfigProxy getConfigProxy(ClassLoader cl);

    @Sensitive
    public default <T> T getConfigValue(ClassLoader cl, String propertyName, Class<T> propertyType) throws IllegalArgumentException, NoSuchElementException {
        if (isAcceptableMpConfigProperty(propertyName)) {
            Optional<T> value = getConfigProxy(cl).getOptionalValue(propertyName, propertyType);
            if (value != null && value.isPresent()) {
                return value.get();
            }
        }
        return null;
    }
```

and each implementation would have identical implementation replacing the existing protected getConfig(ClassLoader) method that it currently has as:
```
    @Override
    public MpConfigProxy getConfigProxy(ClassLoader cl) {
        Config config = cl != null ? ConfigProvider.getConfig(cl) : ConfigProvider.getConfig();

        return new MpConfigProxy() {
            @Override
            public <T> Optional<T> getOptionalValue(String propertyName, Class<T> propertyType) {
                return config.getOptionalValue(propertyName, propertyType);
            }
        };
    }
```

So there would still be duplicated code, but much less of it.  The getConfigProxy method would not exactly be identical because they are each referencing different MP Config versions.

Since you approved the PR, I would prefer that Joe is able to merge the PR and then I or Joe can make a follow on PR with the above code in it plus the new method that was created with this PR.  Maybe can also remove the getConfigValue() method since it isn't used any longer after this PR's change.